docs/spec/index.md

```diff
@@ -133,9 +136,28 @@ Events that don't have `trip_id` set may overlap in time with any other events.
 
 Because some events may overlap in time, it may not be possible to choose a single order for events within a run that's correct for all uses. Producers should use `event_sequence` to define a reasonable order. If a consumer cares about exactly how overlapping events are ordered, they should sort based on the time fields and `event_type` instead.
 
+#### Run ID Uniqueness
+
+Run IDs may be non-unique. E.g. E.g. there may be a "Run 100" on both Weekday and Weekend service. There may even be a Run 100 on both `garage1-weekday` and `garage2-weekday` services, happening on the same day. Runs are uniquely referred to by a `(service_id, run_id)` pair. This is why the service ID is required on this file and other files that refer to run IDs.
+
 #### `run_events` Notes
 
 - Multiple `run_event`s may refer to the same `trip_id`, if multiple employees work on that trip.
 - Events may have gaps between the end time of one event and the start time of the next. e.g. if an operator's layovers aren't represented by an event.
 - `start_time` may equal `end_time` for an event that's a single point in time (such as a report time) without any duration.
 - Recommended sort order: `service_id`, `run_id`, `event_sequence`.
+
+### `employee_run_dates.txt`
+
+Describes which employees are scheduled to which runs on which dates.
+
+This file should represent the schedule after holidays, vacation, and other scheduled exceptions have been applied.
+
+Primary Key: `*`
```

Comment:
The primary key being `*` suggests that multiple employees could be schedule for the same run and service_id on the same day. While we know this situation occurs in reality when there are service problems, is there a case for scheduling this way? We suggest the primary key be `(date, run_id, service_id)`.

Reply:
Yeah, I think this is good. My original thought was that if two employees really are doing the same thing all day (maybe like 2 fare collectors who work the same trains), they could share a run number and the producer could skip some rows. But consumers knowing that there's at most one employee per run is more valuable.

I've made the change.

Reply:
I'd request to revert this change for the original reason cited by @skyqrose: on the railroad front, it is common to have multiple employees assigned to the same run.

While some may pre-specify by craft, others include dual-certified personnel whereby two employees work the same run and are required to vary up who works as the engineer vs conductor on each trip. Likewise, other operations include two matched crew members who work identical shifts in the same function.

In practice, I suspect most consumers can treat `(date, run_id, service_id)` as the primary key, but retaining the flexibility for those of us who schedule that way would be ideal.

Reply:
Ah, I didn't realize that there were schedules like that in practice. Given that this is a real situation, I'll revert it.